### PR TITLE
Fix SmoothL1Criterion and SoftmaxWithCriterion

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?

Fix SmoothL1Criterion and SoftmaxWithCriterion

## How was this patch tested?

unit test

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SmoothL1CriterionWithWeights.scala [hunk 1]
```diff
@@ -135,32 +136,33 @@ class SmoothL1CriterionWithWeights[@specialized(Float, Double) T: ClassTag]
...
     }
-    val alpha = if (num > 0) {
-      ev.fromType(1.0 / num)
+
+    gradInput.resizeAs(diff).copy(diff)
+    if (num > 0) {
+      gradInput.div(ev.fromType(num))
     } else {
-      ev.fromType(input.size(1))
+      gradInput.div(ev.fromType(input.size(1)))
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SmoothL1CriterionWithWeights.scala]: mul?

Comment (SeaOfOcean) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SmoothL1CriterionWithWeights.scala]: The old logic is not correct

Comment (qiuxin2012) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SmoothL1CriterionWithWeights.scala]: OK

**Comment by qiuxin2012** [comment]
LGTM

**Comment by SeaOfOcean** [comment]
pr validate pass

**Pull request merged**